### ISSUE TITLE
Implement possibility to train on huge datasets from python

### ISSUE THREAD

**Issue by annaveronika** [catboost/catboost#1116]

**Comment by andrey-khropov** [comment]
This functionality has been implemented by commit c62fae0bc2ea80d2cb32b73fdc9011ced6724e6f and will be included in the next release.

It is implemented by `catboost.utils.quantize` function.

A simple example how to quantize input data on the fly is:

```py
import catboost
import catboost.utils

quantized_pool = catboost.utils.quantize('/home/user/my_dataset.tsv')
model = catboost.CatBoostRegressor()
model.fit(quantized_pool)
```

If you have separate train and eval or test datasets you have to use the same borders to quantize eval and test datasets as was used for train (for compatibility).

Example:

```py
import tempfile

import catboost
import catboost.utils

with tempfile.NamedTemporaryFile(mode='w') as borders_file:
    quantized_train = catboost.utils.quantize('/home/user/my_dataset/train.tsv')
    quantized_train.save_quantization_borders(borders_file.name)
    quantized_eval = catboost.utils.quantize('/home/user/my_dataset/eval.tsv', input_borders=borders_file.name)
    model = catboost.CatBoostRegressor()
    model.fit(quantized_train, eval_set=quantized_eval)
```

**Comment by fingoldo** [comment]
> This functionality has been implemented by commit [c62fae0](https://github.com/catboost/catboost/commit/c62fae0bc2ea80d2cb32b73fdc9011ced6724e6f) and will be included in the next release.
> 
> It is implemented by `catboost.utils.quantize` function.
> 
> A simple example how to quantize input data on the fly is:
> 
> ```python
> import catboost
> import catboost.utils
> 
> quantized_pool = catboost.utils.quantize('/home/user/my_dataset.tsv')
> model = catboost.CatBoostRegressor()
> model.fit(quantized_pool)
> ```
> 
> If you have separate train and eval or test datasets you have to use the same borders to quantize eval and test datasets as was used for train (for compatibility).
> 
> Example:
> 
> ```python
> import tempfile
> 
> import catboost
> import catboost.utils
> 
> with tempfile.NamedTemporaryFile(mode='w') as borders_file:
>     quantized_train = catboost.utils.quantize('/home/user/my_dataset/train.tsv')
>     quantized_train.save_quantization_borders(borders_file.name)
>     quantized_eval = catboost.utils.quantize('/home/user/my_dataset/eval.tsv', input_borders=borders_file.name)
>     model = catboost.CatBoostRegressor()
>     model.fit(quantized_train, eval_set=quantized_eval)
> ```

Can the sentence  **"If you have separate train and eval or test datasets you have to use the same borders to quantize eval and test datasets as was used for train (for compatibility)"** and this code snippet be added to the docs on quantization, please? I ended up spending a lot of time quantizing different data batches only to find out that I can't score the next pool with model trained on a previous pool: _Feature 0: inconsistent borders, quantized vs model_.

**Comment by antipisa** [comment]
@andrey-khropov How do you merge several train pools that are quantized with the same borders?
